dot/sync/chain_processor.go

```diff
@@ -191,6 +186,10 @@ func (s *chainProcessor) processBlockData(bd *types.BlockData) error {
 		s.handleJustification(bd.Header, *bd.Justification)
 	}
 
+	if err := s.blockState.CompareAndSetBlockData(bd); err != nil {
+		return fmt.Errorf("failed to compare and set data: %w", err)
+	}
```

Comment:
Why moving `CompareAndSetBlockData`? There must be a reason but it's not totally mentioned in the description 🤔

Reply:
yes, sorry I should have mentioned. this should definitely be happening after the block checks happen, otherwise we might be storing some data in the database and the block turns out to be invalid, and then the extra block data is still stored.